corehq/apps/enterprise/resumable_iterator_wrapper.py

```diff
@@ -0,0 +1,31 @@
+class ResumableIteratorWrapper:
+    def __init__(self, sequence, get_element_properties_fn=None):
+        self.it = iter(sequence)
+        self.prev_element = None
+        self.iteration_started = False
+        self.is_complete = False
+
+        self.get_element_properties_fn = get_element_properties_fn
+        if not self.get_element_properties_fn:
+            self.get_element_properties_fn = lambda ele: {'value': ele}
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        self.iteration_started = True
+        try:
+            self.prev_element = next(self.it)
+        except StopIteration:
+            self.is_complete = True
+            raise
+
+        return self.prev_element
+
+    def get_next_query_params(self):
+        if self.is_complete:
+            return None
+        if not self.iteration_started:
+            return {}
```

Comment:
Is there a meaningful difference between these return values on the client? If not, would it work to return an empty dict if the iteration is complete? It would simplify the return type (always returns a dict).

Reply:
The goal was to be able to use this value to determine whether the iterator was complete or not, but as you recommended, I've removed this class now